includes/category-management/index.php

```diff
@@ -20,18 +20,18 @@ function event_espresso_categories_config_mnu() {
 				while (list($key, $value) = each($_POST['checkbox'])):
 					$del_id = $key;
 					//Delete category data
-					$sql = "DELETE FROM " . EVENTS_CATEGORY_TABLE . " WHERE id='$del_id'";
+					$sql = "DELETE FROM " . EVENTS_CATEGORY_TABLE . " WHERE id='" . (int)$del_id . "'";
```

Comment:
Do we need $wpdp->prepare here?

I'm asking because this is very similar to some code I'm updating, as its only using the ID I would use something like:

`$wpdb->delete(EVENTS_CATEGORY_TABLE, array('id' => $del_id ), array('%d'));`

Is that no good?

Reply:
Fixed

Reply:
> Do we need $wpdp->prepare here?
It's not strictly required. The query was safe with the typecasting `$del_id` to an int; it was just unorthodox, and easier to make a mistake that would make in unsafe (eg if you added a new WHERE condition, you might forget to similarly escape the other value.)

> Is that no good?

As I understand it, `$wpdb->delete()` internally uses `$wpdb->prepare()`. So `$wpdb->delete()` is fine too. (The point is: anytime you're passing in an SQL statement to a `$wpdb` method that has some variables in it, it should usually be ran through `$wpdb->prepare()`)